File: SDKs/rudder.example.local/13/java/lib/src/main/java/openapisdk/Techniques.java
package openapisdk;

import openapisdk.utils.HTTPClient;
import openapisdk.utils.HTTPRequest;
import java.net.http.HttpResponse;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.nio.charset.StandardCharsets;

public class Techniques {
	private HTTPClient _defaultClient;
	private HTTPClient _securityClient;
	private String _serverUrl;
	private String _language;
	private String _sdkVersion;
	private String _genVersion;

	public Techniques(HTTPClient defaultClient, HTTPClient securityClient, String serverUrl, String language, String sdkVersion, String genVersion) {
		this._defaultClient = defaultClient;
		this._securityClient = securityClient;
		this._serverUrl = serverUrl;
		this._language = language;
		this._sdkVersion = sdkVersion;
		this._genVersion = genVersion;
	}
	
	
    /**
     * listTechniqueVersionDirectives - List all directives based on a version of a technique
     *
     * List all directives based on a version of a technique
    **/
    public openapisdk.models.operations.ListTechniqueVersionDirectivesResponse listTechniqueVersionDirectives(openapisdk.models.operations.ListTechniqueVersionDirectivesRequest request) throws Exception {
        String baseUrl = this._serverUrl;
        String url = openapisdk.utils.Utils.generateURL(baseUrl, "/techniques/{techniqueName}/{techniqueVersion}/directives", request.pathParams);
        
        HTTPRequest req = new HTTPRequest();
        req.setMethod("GET");
        req.setURL(url);
        
        
        HTTPClient client = this._securityClient;
        
        HttpResponse<byte[]> httpRes = client.send(req);

        String contentType = httpRes.headers().allValues("Content-Type").get(0);

        openapisdk.models.operations.ListTechniqueVersionDirectivesResponse res = new openapisdk.models.operations.ListTechniqueVersionDirectivesResponse() {{
            listTechniqueVersionDirectives200ApplicationJSONObject = null;
        }};
        res.statusCode = Long.valueOf(httpRes.statusCode());
        res.contentType = contentType;
        
        if (httpRes.statusCode() == 200) {
            if (openapisdk.utils.Utils.matchContentType(contentType, "application/json")) {
                ObjectMapper mapper = new ObjectMapper();
                mapper.findAndRegisterModules();
                openapisdk.models.operations.ListTechniqueVersionDirectives200ApplicationJson out = mapper.readValue(new String(httpRes.body(), StandardCharsets.UTF_8), openapisdk.models.operations.ListTechniqueVersionDirectives200ApplicationJson.class);
                res.listTechniqueVersionDirectives200ApplicationJSONObject = out;
            }
        }

        return res;
    }
	
	
    /**
     * listTechniques - List all techniques
     *
     * List all technique with their versions
    **/
    public openapisdk.models.operations.ListTechniquesResponse listTechniques() throws Exception {
        String baseUrl = this._serverUrl;
        String url = openapisdk.utils.Utils.generateURL(baseUrl, "/techniques");
        
        HTTPRequest req = new HTTPRequest();
        req.setMethod("GET");
        req.setURL(url);
        
        
        HTTPClient client = this._securityClient;
        
        HttpResponse<byte[]> httpRes = client.send(req);

        String contentType = httpRes.headers().allValues("Content-Type").get(0);

        openapisdk.models.operations.ListTechniquesResponse res = new openapisdk.models.operations.ListTechniquesResponse() {{
            listTechniques200ApplicationJSONObject = null;
        }};
        res.statusCode = Long.valueOf(httpRes.statusCode());
        res.contentType = contentType;
        
        if (httpRes.statusCode() == 200) {
            if (openapisdk.utils.Utils.matchContentType(contentType, "application/json")) {
                ObjectMapper mapper = new ObjectMapper();
                mapper.findAndRegisterModules();
                openapisdk.models.operations.ListTechniques200ApplicationJson out = mapper.readValue(new String(httpRes.body(), StandardCharsets.UTF_8), openapisdk.models.operations.ListTechniques200ApplicationJson.class);
                res.listTechniques200ApplicationJSONObject = out;
            }
        }

        return res;
    }
	
	
    /**
     * listTechniquesDirectives - List all directives based on a technique
     *
     * List all directives based on all version of a technique
    **/
    public openapisdk.models.operations.ListTechniquesDirectivesResponse listTechniquesDirectives(openapisdk.models.operations.ListTechniquesDirectivesRequest request) throws Exception {
        String baseUrl = this._serverUrl;
        String url = openapisdk.utils.Utils.generateURL(baseUrl, "/techniques/{techniqueName}/directives", request.pathParams);
        
        HTTPRequest req = new HTTPRequest();
        req.setMethod("GET");
        req.setURL(url);
        
        
        HTTPClient client = this._securityClient;
        
        HttpResponse<byte[]> httpRes = client.send(req);

        String contentType = httpRes.headers().allValues("Content-Type").get(0);

        openapisdk.models.operations.ListTechniquesDirectivesResponse res = new openapisdk.models.operations.ListTechniquesDirectivesResponse() {{
            listTechniquesDirectives200ApplicationJSONObject = null;
        }};
        res.statusCode = Long.valueOf(httpRes.statusCode());
        res.contentType = contentType;
        
        if (httpRes.statusCode() == 200) {
            if (openapisdk.utils.Utils.matchContentType(contentType, "application/json")) {
                ObjectMapper mapper = new ObjectMapper();
                mapper.findAndRegisterModules();
                openapisdk.models.operations.ListTechniquesDirectives200ApplicationJson out = mapper.readValue(new String(httpRes.body(), StandardCharsets.UTF_8), openapisdk.models.operations.ListTechniquesDirectives200ApplicationJson.class);
                res.listTechniquesDirectives200ApplicationJSONObject = out;
            }
        }

        return res;
    }
	
}

File: SDKs/rudder.example.local/13/typescript/src/sdk/techniques.ts
import { AxiosInstance, AxiosRequestConfig, AxiosResponse } from "axios";
import * as operations from "./models/operations";
import * as utils from "../internal/utils";

export class Techniques {
  _defaultClient: AxiosInstance;
  _securityClient: AxiosInstance;
  _serverURL: string;
  _language: string;
  _sdkVersion: string;
  _genVersion: string;

  constructor(defaultClient: AxiosInstance, securityClient: AxiosInstance, serverURL: string, language: string, sdkVersion: string, genVersion: string) {
    this._defaultClient = defaultClient;
    this._securityClient = securityClient;
    this._serverURL = serverURL;
    this._language = language;
    this._sdkVersion = sdkVersion;
    this._genVersion = genVersion;
  }
  
  /**
   * listTechniqueVersionDirectives - List all directives based on a version of a technique
   *
   * List all directives based on a version of a technique
  **/
  listTechniqueVersionDirectives(
    req: operations.ListTechniqueVersionDirectivesRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.ListTechniqueVersionDirectivesResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.ListTechniqueVersionDirectivesRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = utils.generateURL(baseURL, "/techniques/{techniqueName}/{techniqueVersion}/directives", req.pathParams);
    
    const client: AxiosInstance = this._securityClient!;
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...config,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.ListTechniqueVersionDirectivesResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `application/json`)) {
                res.listTechniqueVersionDirectives200ApplicationJSONObject = httpRes?.data;
            }
            break;
        }

        return res;
      })
  }

  
  /**
   * listTechniques - List all techniques
   *
   * List all technique with their versions
  **/
  listTechniques(
    config?: AxiosRequestConfig
  ): Promise<operations.ListTechniquesResponse> {
    const baseURL: string = this._serverURL;
    const url: string = baseURL.replace(/\/$/, "") + "/techniques";
    
    const client: AxiosInstance = this._securityClient!;
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...config,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.ListTechniquesResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `application/json`)) {
                res.listTechniques200ApplicationJSONObject = httpRes?.data;
            }
            break;
        }

        return res;
      })
  }

  
  /**
   * listTechniquesDirectives - List all directives based on a technique
   *
   * List all directives based on all version of a technique
  **/
  listTechniquesDirectives(
    req: operations.ListTechniquesDirectivesRequest,
    config?: AxiosRequestConfig
  ): Promise<operations.ListTechniquesDirectivesResponse> {
    if (!(req instanceof utils.SpeakeasyBase)) {
      req = new operations.ListTechniquesDirectivesRequest(req);
    }
    
    const baseURL: string = this._serverURL;
    const url: string = utils.generateURL(baseURL, "/techniques/{techniqueName}/directives", req.pathParams);
    
    const client: AxiosInstance = this._securityClient!;
    
    
    const r = client.request({
      url: url,
      method: "get",
      ...config,
    });
    
    return r.then((httpRes: AxiosResponse) => {
        const contentType: string = httpRes?.headers?.["content-type"] ?? "";

        if (httpRes?.status == null) throw new Error(`status code not found in response: ${httpRes}`);
        const res: operations.ListTechniquesDirectivesResponse = {statusCode: httpRes.status, contentType: contentType};
        switch (true) {
          case httpRes?.status == 200:
            if (utils.matchContentType(contentType, `application/json`)) {
                res.listTechniquesDirectives200ApplicationJSONObject = httpRes?.data;
            }
            break;
        }

        return res;
      })
  }

}
